From 252dbed4c82524c20b3564f560422633aa6cb369 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@gmail.com>
Date: Wed, 3 Apr 2024 10:15:47 -0400
Subject: [PATCH] aufs: adjust for v6.9+

The following commit drops SLAB_MEM_SPREAD, so we remove it
from aufs.

commit cdeeaaba174886aa6c1ff4c0c5449c5066dbe82f
Author: Vlastimil Babka <vbabka@suse.cz>
Date:   Fri Feb 23 19:27:17 2024 +0100

    mm, slab: deprecate SLAB_MEM_SPREAD flag

    The SLAB_MEM_SPREAD flag used to be implemented in SLAB, which was
    removed.  SLUB instead relies on the page allocator's NUMA policies.
    Change the flag's value to 0 to free up the value it had, and mark it
    for full removal once all users are gone.

    Reported-by: Steven Rostedt <rostedt@goodmis.org>
    Closes: https://lore.kernel.org/all/20240131172027.10f64405@gandalf.local.home/
    Reviewed-and-tested-by: Xiongwei Song <xiongwei.song@windriver.com>
    Reviewed-by: Chengming Zhou <chengming.zhou@linux.dev>
    Reviewed-by: Roman Gushchin <roman.gushchin@linux.dev>
    Acked-by: David Rientjes <rientjes@google.com>
    Signed-off-by: Vlastimil Babka <vbabka@suse.cz>

Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 fs/aufs/module.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/aufs/module.h b/fs/aufs/module.h
index a9d21d3e5414..784c830d084d 100644
--- a/fs/aufs/module.h
+++ b/fs/aufs/module.h
@@ -129,7 +129,7 @@ enum {
 
 extern struct kmem_cache *au_cache[AuCache_Last];
 
-#define AuCacheFlags		(SLAB_RECLAIM_ACCOUNT | SLAB_MEM_SPREAD)
+#define AuCacheFlags		(SLAB_RECLAIM_ACCOUNT)
 #define AuCache(type)		KMEM_CACHE(type, AuCacheFlags)
 #define AuCacheCtor(type, ctor)	\
 	kmem_cache_create(#type, sizeof(struct type), \
-- 
2.39.2

